### ISSUE TITLE
rust: Generated resource bindings can conflict with resource methods

### ISSUE THREAD

**Issue by lann** [bytecodealliance/wit-bindgen#1059]
```wit
world example {
  resource repro {
    constructor();
    handle: func();
  }
}
```

generates two conflicting `Repro::handle` methods; one `#[doc(hidden)]` internal method and the actual resource method. Same would presumably be true for `take_handle` and (`static`) `from_handle`.

**Comment by alexcrichton** [comment]
Oops, thanks! For purely internal functions we should probably rename them to `_foo` or similar, but otherwise we'll probably need to "mangle" WIT-based names that conflict.

**Comment by lann** [comment]
Do they need to be methods? Could they just be regular (associated) functions?

**Comment by alexcrichton** [comment]
Good point yeah, `doc(hidden)` items can also be associated functions. Using `_` as a prefix for those I think would also be reasonable

**Comment by lann** [comment]
I guess bindgen will never generate a `_` prefix so it doesn't really matter :shrug:
